block/manager.go

```diff
@@ -1165,4 +1162,13 @@ func updateState(s *types.State, res *abci.ResponseInitChain) {
 	// We update the last results hash with the empty hash, to conform with RFC-6962.
 	s.LastResultsHash = merkle.HashFromByteSlices(nil)
 
+	if len(res.Validators) > 0 {
```

Comment:
Isn't it a problem if there are zero validators? 
If so, this should be reverse and an error returned instead of indenting all this under the if condition.